File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    
    want_dict = {}
    for i in range(len(want)):
        want_dict[want[i]] = number[i]
    
    for i in range(len(discount) - sum(number) + 1):
        temp_dict = want_dict.copy()
        for j in range(sum(number)):
            temp_dict[discount[i + j]] = temp_dict.get(discount[i + j], 0) - 1
            
            if temp_dict[discount[i + j]] == 0:
                del temp_dict[discount[i + j]]
            
        if not temp_dict:
            answer += 1
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    
    unordered_map<string, int> wantMap;
    
    for (int i = 0; i < want.size(); i++) {
        wantMap[want[i]] = number[i];
    }
    
    for (int i = 0; i < discount.size() - 9; i++) {
        unordered_map<string, int> discountMap;
        
        for (int j = 0; j < 10; j++) {
            discountMap[discount[i + j]] += 1;
        }
        
        if (wantMap == discountMap) answer++;
    }
    
    return answer;
}